### PR TITLE
{lib}[GCCcore/12.3.0] Highway 1.0.4 fix for failing test on zen4

### PR DESCRIPTION
**Pull request by trz42**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/h/Highway/Highway-1.0.4-zen4-fix-TruncateTo-bug.patch [hunk 1]
```diff
@@ -0,0 +1,44 @@
+diff --git a/hwy/ops/x86_256-inl.h b/hwy/ops/x86_256-inl.h
```
Comment (boegel) [inline easybuild/easyconfigs/h/Highway/Highway-1.0.4-zen4-fix-TruncateTo-bug.patch:1]: @trz42 Can you add a description on top of the patch to explain why this is required (fix failing test on AMD Genoa a.k.a. Zen4 systems), including a pointer to https://github.com/google/highway/issues/1913, and maybe also a bit of info on how you trimmed it down to just this minimal patch + add yourself as patch author.

Keep in mind that this implies updating the SHA256 checksum for the patch file in the easyconfig

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
@boegelbot please test @ generoso
EB_ARGS="--installpath /tmp/20942"

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen3
EB_ARGS="--installpath /tmp/20942"

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node4214.shinx.os - Linux RHEL 9.2, x86_64, AMD EPYC 9654 96-Core Processor, Python 3.9.16
See https://gist.github.com/boegel/eba615933d6dc0ec7916488ad306b710 for a full test report.

File: easybuild/easyconfigs/h/Highway/Highway-1.0.4-zen4-fix-TruncateTo-bug.patch (hunk 1 above)
Comment (akesandgren) [reply easybuild/easyconfigs/h/Highway/Highway-1.0.4-zen4-fix-TruncateTo-bug.patch:1]: @trz42 Ping, please update the patch file with the requested info.
I need this asap :-)

Comment (trz42) [reply easybuild/easyconfigs/h/Highway/Highway-1.0.4-zen4-fix-TruncateTo-bug.patch:16]: @boegel @akesandgren Added a description of the patch file. Let me know if you need additional information.

Note, I haven't tested the updated PR/easyconfig yet.

Comment (trz42) [reply easybuild/easyconfigs/h/Highway/Highway-1.0.4-zen4-fix-TruncateTo-bug.patch:16]: Still builds for Zen4 on top of EESSI. See https://github.com/EESSI/software-layer/pull/627#issuecomment-2255653601

**Comment by boegel** [comment]
@boegelbot please test @ generoso
EB_ARGS="--installpath /tmp/20942"

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node4201.shinx.os - Linux RHEL 9.4, x86_64, AMD EPYC 9654 96-Core Processor (zen4), Python 3.9.18
See https://gist.github.com/boegel/0d17e7d76279134d8c46c04dc5cb365b for a full test report.

**Comment by boegel** [comment]
Going in, thanks @trz42!

**Pull request merged**